src/Cooked/InitialDistribution.hs

```diff
@@ -32,18 +35,36 @@ import PlutusLedgerApi.V3 qualified as Api
 --  >        , (wallet 2 , [ ada 10 ])
 --  >        , (wallet 3 , [ ada 10 <> permanentValue "XYZ" 10])
 --  >        ]
-newtype InitialDistribution = InitialDistribution {unInitialDistribution :: [TxSkelOut]}
+--
+-- Note that initial distribution can lead to payments that would not be
+-- accepted if part of an actual transaction, such as payment without enough ada
+-- to sustain themselves.
+data InitialDistribution where
+  InitialDistribution ::
+    {unInitialDistribution :: [TxSkelOut]} ->
+    InitialDistribution
 
 -- | 5 UTxOs with 100 Ada each, for each of the 'knownWallets'
 instance Default InitialDistribution where
   def = distributionFromList . zip knownWallets . repeat . replicate 5 $ ada 100
 
 instance Semigroup InitialDistribution where
-  i <> j = InitialDistribution $ unInitialDistribution i <> unInitialDistribution j
+  i <> j = InitialDistribution (unInitialDistribution i <> unInitialDistribution j)
 
 instance Monoid InitialDistribution where
   mempty = InitialDistribution mempty
 
+-- | Transform a given initial distribution by ensuring each payment has enough
+-- ada so that the resulting outputs can sustain themselves.
+toInitDistWithMinAda :: InitialDistribution -> InitialDistribution
+toInitDistWithMinAda (InitialDistribution initDist) =
+  case forM initDist $ toTxSkelOutWithMinAda def of
+    Left err -> error $ show err
+    Right initDist' -> InitialDistribution initDist'
```

Comment:
I think we should not make this a partial function. Producing a `Maybe InitialDistribution` or `Either error InitialDistribution` seems fine to me. I propose that we return an `Either error InitialDistribution` but also provide an `unsafeToInitDistWithMinAda` helper for convenience.

In any case, I think the name and documentation should reflect that this is unsafe.